src/main/java/org/jitsi/jicofo/util/DaemonThreadFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jicofo.util;
+
+import java.util.concurrent.*;
+
+/**
+ * A thread factory that creates daemon threads.
+ *
+ * @author Pawel Domas
+ */
+public class DaemonThreadFactory
+    implements ThreadFactory
+{
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Thread newThread(Runnable r)
+    {
+        Thread t = Executors.defaultThreadFactory().newThread(r);
+        // The check is to avoid IllegalStateException if it's already set
```

Comment:
I believe the comment is incorrect.

1. I read the javadoc on Thread#setDaemon(boolean) and it talks about IllegalThreadStateException and SecurityException. I leafed through the source code as well and didn't find a mention of IllegalStateException.

2. As to Thread#isDaemon() being a protection against an exception, the comment makes me think that an IllegalStateException is thrown if t.setDaemon(true) is called after t.isDaemon() is true already. Which is simply not the case because Thread#setDaemon(boolean) throws if the thread is alive already (and if there are security/privilege violations). And Thread#isDaemon() will check whether the thread is alive.

I think it's OK to have the isDaemon check just to not risk a theoretical IllegalThreadStateException (which is practically unlikely to ever happen because Executors.defaultThreadFactory().newThread(r) shouldn't return an alive thread) but the comment is misleading (to me).